apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -99,6 +99,15 @@ public final class RifLoader {
   /** The shared application state. */
   private final PipelineApplicationState appState;
 
+  /** The maximum amount of time in hours we will wait for a job to complete loading its batches. */
+  private final int MAX_BATCH_WAIT_TIME_HOURS = 72;
+
+  /**
+   * Keeps track of a fatal error during loading in one of the batch threads so we know to fail the
+   * job (and kill the pipeline).
+   */
+  private static volatile boolean fatalFailure;
```

Comment:
Would be open to if anyone has a better idea than this that doesnt require rearchitecting the application; basically we need a signal from the subthreads to the main thread that we hit some error and the job has failed, and also allows the thread to complete/release. An exception won't work as it doesnt seem to bubble up to the caller (maybe the batch spliterator is eating it or something?) Not sure what other methods there are for talking between threads that arent over-engineered for this rather simple use-case.

Reply:
this could be an AtomicBoolean; allows for atomic operations. It ensures that reads and writes are thread-safe, meaning that multiple threads can access the same boolean without overriding each other's values.

Reply:
I'll look into that; I used volatile here to try and ensure that the value being read was always up-to-date across threads, since its only being written to with one value. But maybe safer just to use AtomicBoolean since the specific way I implemented it isnt immediately obvious.

Reply:
Adjusted to use AtomicBoolean.